### PR TITLE
Feature/osi 28 passcode UI refactor

### PR DESCRIPTION
**Pull request by Elelan**

### CODE REVIEW

File: Save/Features/Passcode/DialogHelper.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import SwiftUI
+
+class DialogHelper {
+    
+    // MARK: - UIAlertController for UIKit
+    
+    static func showConfirmationAlert(
+        title: String,
+        message: String,
+        confirmButtonTitle: String = "OK",
```
Comment (prathieshna) [inline Save/Features/Passcode/DialogHelper.swift:19]: Localization?

File: Save/Features/Passcode/DialogHelper.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        cancelButtonTitle: String = "Cancel",
```
Comment (prathieshna) [inline Save/Features/Passcode/DialogHelper.swift:20]: Localization?

File: Save/Features/Passcode/DialogHelper.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        
+        viewController.present(alert, animated: true, completion: nil)
+    }
+    
+    // MARK: - SwiftUI Alert
+    
+    static func createConfirmationAlert(
+        title: String,
+        message: String,
+        confirmButtonTitle: String = "OK",
```
Comment (prathieshna) [inline Save/Features/Passcode/DialogHelper.swift:43]: Localization?

File: Save/Features/Passcode/DialogHelper.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        cancelButtonTitle: String = "Cancel",
```
Comment (prathieshna) [inline Save/Features/Passcode/DialogHelper.swift:44]: Localization?

File: Save/Features/Passcode/PasscodeSetup/PasscodeSetupView.swift [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    let onNumberClick: (String) -> Void
+    let onBackspaceClick: () -> Void
+    
+    let onAnimationCompleted: () -> Void
+    
+    @Environment(\.colorScheme) var colorScheme
+    
+    var body: some View {
+        PasscodeContentWrapper(
+            title: isConfirming ? "Confirm Passcode" : "Enter Passcode",
```
Comment (prathieshna) [inline Save/Features/Passcode/PasscodeSetup/PasscodeSetupView.swift:51]: Localization?

**Review by prathieshna — CHANGES_REQUESTED** [review]
Please check the localization for the strings used.

File: Save/Features/Passcode/DialogHelper.swift (hunk 4 above)
Comment (navodaSathsarani) [reply Save/Features/Passcode/DialogHelper.swift:44]: added

File: Save/Features/Passcode/PasscodeSetup/PasscodeSetupView.swift (hunk 5 above)
Comment (navodaSathsarani) [reply Save/Features/Passcode/PasscodeSetup/PasscodeSetupView.swift:51]: added

File: Save/Features/Passcode/DialogHelper.swift (hunk 3 above)
Comment (navodaSathsarani) [reply Save/Features/Passcode/DialogHelper.swift:43]: added

File: Save/Features/Passcode/DialogHelper.swift (hunk 2 above)
Comment (navodaSathsarani) [reply Save/Features/Passcode/DialogHelper.swift:20]: added

File: Save/Features/Passcode/DialogHelper.swift (hunk 1 above)
Comment (navodaSathsarani) [reply Save/Features/Passcode/DialogHelper.swift:19]: added

**Review by prathieshna — APPROVED** [review]
LGTM

**Pull request merged**